File: notification-service/src/main/java/com/tickets/notification_service/notification/Notification.java
package com.tickets.notification_service.notification;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "notifications")
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Enumerated(EnumType.STRING)
    @Column(length = 50, nullable = false)
    private NotificationType type;

    @Column(length = 255)
    private String subject;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String message;

    @Enumerated(EnumType.STRING)
    @Column(length = 50, nullable = false)
    private NotificationStatus status = NotificationStatus.PENDING;

    // Referencia opcional al recurso que disparó la notificación (ej: orderId)
    @Column(name = "reference_id")
    private UUID referenceId;

    @Column(name = "sent_at")
    private LocalDateTime sentAt;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: frontend/eventFlow/src/features/notifications/domain/entities/Notification.ts
export type NotificationType =
  | 'PURCHASE_CONFIRMATION'
  | 'PAYMENT_CONFIRMED'
  | 'EVENT_CANCELLED'
  | 'REFUND_COMPLETED'
  | 'REFUND_FAILED'
  | 'GENERAL'

export type NotificationStatus = 'PENDING' | 'SENT' | 'FAILED'

export interface Notification {
  id: string
  userId: string
  type: NotificationType
  subject: string
  message: string
  status: NotificationStatus
  referenceId: string | null
  sentAt: string | null
  createdAt: string
}
